nifi-nar-bundles/nifi-email-bundle/nifi-email-processors/src/main/java/org/apache/nifi/processors/email/ExtractTNEFAttachments.java

```diff
@@ -0,0 +1,202 @@
+                                        attributes.put(ATTACHMENT_ORIGINAL_FILENAME, originalFlowFileName);
+
+                                        // TODO: Extract Mime Type (HMEF doesn't seem to be able to get this info.
+
+                                        split = session.append(split, new OutputStreamCallback() {
+                                            @Override
+                                            public void process(OutputStream out) throws IOException {
+                                                out.write(attachment.getContents());
+                                            }
+                                        });
+                                        split = session.putAllAttributes(split, attributes);
+                                        attachmentsList.add(split);
+                                    }
+                                } catch (FlowFileHandlingException e) {
+                                    // Something went wrong
+                                    // Removing splits that may have been created
+                                    session.remove(attachmentsList);
+                                    // Removing the original flow from its list
+                                    originalFlowFilesList.remove(originalFlowFile);
+                                    logger.error("Flowfile {} triggered error {} while processing message removing generated FlowFiles from sessions", new Object[]{originalFlowFile, e});
+                                    invalidFlowFilesList.add(originalFlowFile);
+                                }
+                            }
+                        }
+                    }catch (Exception e) {
+                        // Another error hit...
+                        // Removing the original flow from its list
+                        originalFlowFilesList.remove(originalFlowFile);
+                        logger.error("Could not parse the flowfile {} as an email, treating as failure", new Object[]{originalFlowFile, e});
+                        // Message is invalid or triggered an error during parsing
+                        invalidFlowFilesList.add(originalFlowFile);
+                    }
+                }
+        });
+
+        session.transfer(attachmentsList, REL_ATTACHMENTS);
+
+        // As per above code, originalFlowfile may be routed to invalid or
+        // original depending on RFC2822 compliance.
+        session.transfer(invalidFlowFilesList, REL_FAILURE);
+        session.transfer(originalFlowFilesList, REL_ORIGINAL);
+
+        if (attachmentsList.size() > 10) {
+            logger.info("Split {} into {} files", new Object[]{originalFlowFile, attachmentsList.size()});
+        } else if (attachmentsList.size() > 1){
+            logger.info("Split {} into {} files: {}", new Object[]{originalFlowFile, attachmentsList.size(), attachmentsList});
+        }
```

Comment:
It is based on SplitContent

https://github.com/apache/nifi/blob/1bd2cf0d09a7111bcecffd0f473aa71c25a69845/nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/SplitContent.java#L292

But I agree the code is hard to grasp as there are two completely separate tests occurring within the same if then else statement. 

Addressed.